jdbc/src/main/java/org/apache/iotdb/jdbc/Utils.java

```diff
@@ -83,21 +86,95 @@ static IoTDBConnectionParams parseUrl(String url, Properties info)
 
     return params;
   }
-
-  /**
-   * convert row records.
-   */
-  static List<RowRecord> convertRowRecords(TSQueryDataSet tsQueryDataSet,
-      List<String> columnTypeList) {
+  
+  static ByteBuffer convertResultBuffer(TSQueryDataSet tsQueryDataSet,
+      List<String> columnTypeList) throws IOException {
     int rowCount = tsQueryDataSet.getRowCount();
     ByteBuffer byteBuffer = tsQueryDataSet.bufferForValues();
+    DataOutputStream[] dataOutputStreams = new DataOutputStream[rowCount];
+    ByteArrayOutputStream[] byteArrayOutputStreams = new ByteArrayOutputStream[rowCount];
 
     // process time buffer
-    List<RowRecord> rowRecordList = processTimeAndCreateRowRecords(byteBuffer, rowCount);
-
+    processTimestamp(byteBuffer, rowCount, dataOutputStreams, byteArrayOutputStreams);
+    int valueOccupation = 0;
     for (String type : columnTypeList) {
+      TSDataType dataType = TSDataType.valueOf(type);
       for (int i = 0; i < rowCount; i++) {
-        Field field = null;
+        boolean is_empty = BytesUtils.byteToBool(byteBuffer.get());
+        if (is_empty) {
+          dataOutputStreams[i].writeBoolean(true);
+        } else {
+          dataOutputStreams[i].writeBoolean(false);
+          switch (dataType) {
+            case BOOLEAN:
+              boolean booleanValue = BytesUtils.byteToBool(byteBuffer.get());
+              dataOutputStreams[i].writeBoolean(booleanValue);
+              valueOccupation += 1;
```

Comment:
So you convert the `column-wise` buffer to a `row-wise` buffer to replace the `List<RowRecord>` structure, and then get a rowRecord from the `row-wise` buffer each time.

I see that `List<RowRecord>` is replaced by a more compact structure.
But it still looks a little wierd to me. A value derived from the buffer is then written into buffer again... Do you have any opinions @jt2594838?

Reply:
Yes, you are talking about what I am thinking. The buffer fetched from server is `column-wise`. If we can construct the `row-wise` buffer in server, we won't need to convert the buffer again.

Reply:
Good idea! 
I look back on [my development document](https://tower.im/teams/138444/todos/7987/) of `column-wise` buffer and can't find any strong reason against a `row-wise` alternative.